realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1379,9 +1379,8 @@ public RealmAsyncTask executeTransactionAsync(final Transaction transaction,
 
         // If the user provided a Callback then we have to make sure the current Realm has an events looper to deliver
         // the results.
-        if ((onSuccess != null || onError != null)  && !canDeliverNotification) {
-            throw new IllegalStateException("Your Realm is opened from a thread without an event looper." +
-                    " The callback cannot be invoked.");
+        if ((onSuccess != null || onError != null)) {
```

Comment:
@cmelchior I always get confused by boolean logic and negation, should this be & &  or ||?